Makefile

```diff
@@ -7,11 +7,22 @@ build:
 	cargo build --release
 
 .PHONY: test
-test:
-	RUST_LOG=$(LOG_LEVEL) cargo test --all --no-fail-fast -- --nocapture --include-ignored
+test: lint test-unit test-integration
+
+.PHONY: lint 
+lint:
 	cargo clippy --all-targets --all-features -- -D warnings
 	cargo fmt --all -- --check
 
+.PHONY: test-unit
+test-unit:
+	RUST_LOG=$(LOG_LEVEL) cargo test --no-fail-fast -- --skip integration_tests --nocapture --include-ignored
+
+.PHONY: test-integration
+test-integration:
+	RUST_LOG=$(LOG_LEVEL) cargo test integration_tests::test_dependencies --no-fail-fast -- --nocapture
+	RUST_LOG=$(LOG_LEVEL) cargo test integration_tests --no-fail-fast -- --skip integration_tests::test_dependencies --nocapture --include-ignored
```

Comment:
What is the reason we're using this flag: `--skip integration_tests::test_dependencies`?